plugins/kubernetes-backend/api-report.md

```diff
@@ -27,6 +28,8 @@ export interface ClusterDetails {
   // (undocumented)
   authProvider: string;
   // (undocumented)
+  dashboardUrl?: string;
```

Comment:
I'm feeling that we should make the this a function at the cluster locator level, probably just `(k8sObject) => string`. There are a lot of other dashboards that people use rather than the standard one, so I feel this locks into that a bit too much. With it being a function you can keep the current way of configuring the URL and convert it to a function when reading config, but then leave it much more open for other implementations.

Reply:
Hi,
Excellent comment! I also had the feeling that there would be many different kinds of dashboards, and I lazily only provided something for the standard dashboard! :-)
I think that having the dashboardUrl in the config still makes sense and will help configuring Backstage, especially for the happy path.
Adding a resolver function will help for sure!
Also, I was wondering if we should add a 'dashboardProvider' or 'dashboardKind' or 'dashboardApp' property that would contain an enumeration like standard,openshift,foobar...
Then we could have dedicated functions and a mapping in order to help configuring Backstage.
What do you think?

Reply:
Hi,

Sorry about the delay but I was on vacation!

I've refactored the existing code in order to support multiple link formatters and also allow one to register its own formatter.

I've also started adding documentation and I provided the empty formatters for the dashboards other than 'standard', to let people contribute if they need them.

Let me know if the PR meets your expectations now...

P.S. The "open kubernetes dashboard" button does not render properly since the text color of the link (inside the button) has the same color as the button itself!!!!
You told me to change the button to something like this but it seems that there is a regression:
```tsx
import {
  Button as BackstageButton,
  CodeSnippet,
  StructuredMetadataTable,
  Link,
} from '@backstage/core-components';

<BackstageButton
              variant="contained"
              color="primary"
              size="small"
              component={Link}
              to={clusterLink}
            >
              Open Kubernetes Dashboard...
</BackstageButton>
```